### ISSUE TITLE
stdio captured by logging adds spaces

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7852]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#7852|
|Type|enhancement|
|Created|2015-04-15 02:56:24Z|
|Branch|https://github.com/twisted/twisted/tree/log-spaces-7852|

In looking into some other stuff, we've found a regression in new logging.  This program:

```
import sys
from twisted.python.log import startLogging

startLogging(sys.stdout)
print "Hello!"
```

Outputs `Hello!` with spaces, which is rather unexpected:

```
2015-04-14 22:54:31-0400 [-] Log opened.
2015-04-14 22:54:31-0400 [-] H e l l o !
```

<details><summary>Searchable metadata</summary>

```
trac-id__7852 7852
type__enhancement enhancement
reporter__wsanchez wsanchez
priority__highest highest
milestone__None None
branch__branches_log_spaces_7852 branches/log-spaces-7852
branch_author__wsanchez wsanchez
status__closed closed
resolution__fixed fixed
component__core core
keywords__logging__BOOYAH_ logging, BOOYAH!
time__1429066584526250 1429066584526250
changetime__1445217226370668 1445217226370668
version__None None
owner__wsanchez wsanchez

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_1"></a>|@wsanchez commented|
|-|-|

Tagged as highest priority for 15.2.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_2"></a>|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `assigned`|
|-|-|

ralphm has used his magical debugging superpowers and found the problem.  There's a trivial fix, will put on a branch momentarily.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_3"></a>|@wsanchez commented|
|-|-|

(In [44446]) Branch for [#7852](https://github.com/twisted/twisted/issues/7852)

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez commented|
|-|-|

So… the problem that ralph found is that in `LoggingFile.write()`, we capture a file stream (stdout in the above example) and emit a new-style log event with a key called `message` for each line of text.

That then gets forwarded to a legacy log observer, wherein it gets renders with `textFromEventDict`, which has certain expectations of an event key called `message`, specifically that it's a `tuple`, which should be joined by spaces.

This is trivially fixed:

```
-            self.log.emit(self.level, format=u"{message}", message=line)
+            self.log.emit(self.level, format=u"{log_io}", log_io=line)
```

This does not fix the problem that the `message` key becomes effectively reserved as an artifact of the old logging system still being in play.  I do think we need to fix that also, but that specific problem isn't a regression, and the above is a big regression, so I propose we fix the above here and separately deal with the `message` key problem.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @ralphm<br>@wsanchez set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_7"></a>|@wsanchez commented|
|-|-|

[#7855](https://github.com/twisted/twisted/issues/7855) addresses the `message` key issue, which is the root cause here.  Still want to merge this branch to rename the key we use here, as a cleanup item.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_8"></a>|@ralphm set owner to @wsanchez<br>@ralphm set status to `new`|
|-|-|

This looks good to me. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_9"></a>|@wsanchez set status to `closed`|
|-|-|

(In [44492]) Merge branches/log-spaces-7852: Change `messages` key to `log_io` for events generated by `LoggingFile`.

Author: wsanchez
Reviewer: ralphm
Fixes: [#7852](https://github.com/twisted/twisted/issues/7852)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

Ticket retargeted after milestone deleted
